test/Ocelot.UnitTests/DownstreamRouteFinder/DownstreamRouteFinderTests.cs

```diff
@@ -66,6 +70,7 @@ public void should_return_highest_priority_when_first()
                 }, string.Empty, serviceProviderConfig))
                 .And(x => x.GivenTheUrlMatcherReturns(new OkResponse<UrlMatch>(new UrlMatch(true))))
                 .And(x => x.GivenTheUpstreamHttpMethodIs("Post"))
+                .And(x => x.GivenEmptyRequestHeaders())

```

Comment:
This and other tests below are not your ones.
Why should we add this extra setup call?

Reply:
Because the call sets up a local variable that is also referenced by all these other tests. The code being tested expects the variable to at least not be null, hence why I at least set the variable to an empty instance.

Reply:
OK, but are there another possible design/solutions?

Reply:
You mean other than flat-out initializing the reference in a constructor? My goal was to follow the pattern established by the other methods in the test, first and foremost. I have added functionality to the DownstreamRouteFinder, even made a change to the interface method, so any test that is calling this method had to be updated to include the new argument.

Reply:
We can propose another pattern, code snippets which allow to write less code.
Isn't it an argument "writing less code", "not changing irrelevant code blocks/methods"?

Reply:
It's getting frustrating if I'm being honest. I have written the code trying to follow the established patterns as closely as possible. Now I just want to fix any potential bugs and efficiency issues, not philosophize about aspects of the code that should have been taken into account years ago.

Reply:
**Fixed** in https://github.com/ThreeMammals/Ocelot/pull/1684/commits/90ee5e8baa8e00a965e7ddbb65f23592d7eed7be